statime-linux/src/clock/mod.rs

```diff
@@ -7,55 +7,49 @@ use statime::{Clock, Duration, Time, TimePropertiesDS};
 
 #[derive(Debug, Clone)]
 pub struct LinuxClock {
-    pub clock: clock_steering::unix::UnixClock,
+    clock: clock_steering::unix::UnixClock,
+    is_tai: bool,
 }
 
 impl LinuxClock {
-    pub const CLOCK_REALTIME: Self = Self {
-        clock: UnixClock::CLOCK_REALTIME,
+    pub const CLOCK_TAI: Self = Self {
+        clock: UnixClock::CLOCK_TAI,
+        is_tai: true,
     };
 
     pub fn open(path: impl AsRef<Path>) -> std::io::Result<Self> {
         let clock = UnixClock::open(path)?;
 
-        Ok(Self { clock })
-    }
-}
-
-impl clock_steering::Clock for LinuxClock {
-    type Error = clock_steering::unix::Error;
-
-    fn now(&self) -> Result<clock_steering::Timestamp, Self::Error> {
-        self.clock.now()
-    }
-
-    fn resolution(&self) -> Result<clock_steering::Timestamp, Self::Error> {
-        self.clock.resolution()
-    }
-
-    fn set_frequency(&self, frequency: f64) -> Result<clock_steering::Timestamp, Self::Error> {
-        self.clock.set_frequency(frequency)
+        Ok(Self {
+            clock,
+            is_tai: false,
+        })
     }
 
-    fn step_clock(&self, offset: TimeOffset) -> Result<clock_steering::Timestamp, Self::Error> {
-        self.clock.step_clock(offset)
+    pub fn system_offset(&self) -> Result<(Time, Time, Time), clock_steering::unix::Error> {
+        self.clock.system_offset().map(|(mut t1, t2, mut t3)| {
+            let tai_offset = UnixClock::CLOCK_REALTIME.get_tai().unwrap();
+            t1.seconds += tai_offset as libc::time_t;
+            t3.seconds += tai_offset as libc::time_t;
+            (
+                clock_timestamp_to_time(t1),
+                clock_timestamp_to_time(t2),
+                clock_timestamp_to_time(t3),
+            )
+        })
```

Comment:
this function is confusing. There is a lot of subtle `tai` stuff going on, and I believe you but do not fully understand why this is happening?
```suggestion
    /// Some description of what we return here
    pub fn system_offset(&self) -> Result<(Time, Time, Time), clock_steering::unix::Error> {
        // what do we get back?
        let (mut t1, t2, mut t3) = self.clock.system_offset()?;
        let tai_offset = UnixClock::CLOCK_REALTIME.get_tai()?; // function should be `get_tai_offset`

        t1.seconds += tai_offset as libc::time_t;
        t3.seconds += tai_offset as libc::time_t;

        Ok((
            clock_timestamp_to_time(t1),
            clock_timestamp_to_time(t2),
            clock_timestamp_to_time(t3),
        ))
```

Reply:
Added explanation. for the get_tai stuff see above